pipelines/src/pipelines/utils/trigger_pipeline.py

```diff
@@ -84,10 +90,22 @@ def trigger_pipeline(
         type=str,
     )
 
+    parser.add_argument(
+        "--wait",
+        help="Wait for the pipeline to finish running",
+        type=str,
+    )
     # Get commandline args
     args = parser.parse_args()
 
+    if args.wait.lower() == "true":
+        wait = True
+    elif args.wait.lower() != "false":
+        raise ValueError("wait variable must be 'true' or 'false'")
+    wait = False
```

Comment:
@becky-dt can you please check for similar logic elsewhere in this repo? would like to follow the same logic or even import a reusable function for parsing boolean flags from the command-line. thanks!

Reply:
@felix-datatonic There are a couple more booleans in the Makefile but in this case the logic is all run in the makefile (unlike above where the wait param is passed into the python function)
Do we want to add similar logic in the makefile too to catch ValueErrors? and if so is there a good way to raise errors in bash
(let me know if this doesnt make sense and needs more explaining!)

Reply:
@roberta-dt and I added this logic to the makefile in the latest commit :)

Reply:
77f2584

Reply:
@browningjp-datatonic can you please take a look at the change we made to the run command?

Reply:
The other place where similar logic is used in Python is for `ENABLE_PIPELINE_CACHING` - however this is a bit different, because it's not just `True`/`False`, but `True`/`False`/`None` (`None` default)

The `compile=true` and `build=true` flags are implemented in the Makefile itself as you say, and it's not simple to replicate the logic exactly.

For consistency in behaviour, I would suggest to modify your Python logic here to:

```python
    if args.wait == "true":
        wait = True
    elif args.wait != "false":
        raise ValueError("wait variable must be 'true' or 'false'")
    wait = False
```

(value from user must be `true` or `false` exactly - case sensitive)

Reply:
59305a6